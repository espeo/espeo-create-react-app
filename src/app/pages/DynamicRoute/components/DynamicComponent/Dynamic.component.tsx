import React, { PureComponent } from 'react';

export default class DynamicComponent extends PureComponent {
  public render() {
    return (
      <div className="dynamic-component">
        <p className="example-content">Example content</p>
      </div>
    );
  }
}
