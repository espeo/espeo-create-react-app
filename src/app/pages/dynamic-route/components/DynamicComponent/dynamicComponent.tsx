import React, { PureComponent } from 'react';

export default class DynamicComponent extends PureComponent {
  public render() {
    return <div>Example content</div>;
  }
}
