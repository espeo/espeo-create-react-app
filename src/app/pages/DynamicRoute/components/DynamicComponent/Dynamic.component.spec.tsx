import renderer from 'react-test-renderer';
import React from 'react';
import { compose } from 'redux';
import { shallow, mount } from 'enzyme';

import { withProviders } from '@core/tests/theme-helpers';

import DynamicComponent from './Dynamic.component';

const WrappedComponent = compose(withProviders)(DynamicComponent);

describe('Dynamic component test suite', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<WrappedComponent />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should mount component with children', () => {
    const text = 'Example content';
    const wrapper = mount(<WrappedComponent />);
    expect(wrapper.find('.example-content').text()).toEqual(text);
  });

  it('should render dynamic component', () => {
    const wrapper = shallow(<DynamicComponent />);
    expect(wrapper.find('.dynamic-component')).toHaveLength(1);
  });
});
