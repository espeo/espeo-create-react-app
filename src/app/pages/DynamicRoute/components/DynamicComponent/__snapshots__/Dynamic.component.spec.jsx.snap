// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dynamic component test suite should match snapshot 1`] = `
<div
  className="dynamic-component"
>
  <p
    className="example-content"
  >
    Example content
  </p>
</div>
`;
